packages/harmony/src/components/button/FollowButton/FollowButton.tsx

```diff
@@ -0,0 +1,122 @@
+import { useState, useCallback, ChangeEvent } from 'react'
+
+import { useTheme, type CSSObject } from '@emotion/react'
+
+import type { IconComponent } from 'components/icon'
+import { Flex } from 'components/layout/Flex'
+import { Text } from 'components/text/Text'
+import { useControlled } from 'hooks/useControlled'
+import { IconUserFollowing, IconUserFollow, IconUserUnfollow } from 'icons'
+
+import type { FollowButtonProps } from './types'
+
+const messages = {
+  follow: 'Follow',
+  following: 'Following',
+  unfollow: 'Unfollow'
+}
+
+/**
+ * Special button for following or unfollowing a user.
+ */
+export const FollowButton = (props: FollowButtonProps) => {
+  const {
+    variant = 'default',
+    following = false,
+    onUnfollow,
+    onFollow,
+    size = 'default',
+    ...inputProps
+  } = props
+  const [value, setValueState] = useControlled({
+    componentName: 'FollowButton',
+    controlledProp: following,
+    defaultValue: undefined,
+    stateName: 'following'
+  })
+
+  // Track hover manually to swap text and icon
+  const [isHovering, setIsHovering] = useState(false)
+  const handleMouseEnter = useCallback(() => {
+    setIsHovering(true)
+  }, [setIsHovering])
+  const handleMouseLeave = useCallback(() => {
+    setIsHovering(false)
+  }, [setIsHovering])
+
+  const handleChange = useCallback(
+    (e: ChangeEvent<HTMLInputElement>) => {
```

Comment:
do want to give e to the onFollow/onUnfollow? otherwise maybe we just remove it?